HW-6/pack_hw_6/chess.py

```diff
@@ -0,0 +1,88 @@
+# 1. Добавьте в пакет, созданный на семинаре шахматный модуль. Внутри него напишите код, решающий задачу о 8 ферзях.
+# Известно, что на доске 8×8 можно расставить 8 ферзей так, чтобы они не били друг друга.
+# Вам дана расстановка 8 ферзей на доске, определите, есть ли среди них пара бьющих друг друга.
+# Программа получает на вход восемь пар чисел, каждое число от 1 до 8 - координаты 8 ферзей.
+# Если ферзи не бьют друг друга верните истину, а если бьют - ложь.
+# 2. Напишите функцию в шахматный модуль. Используйте генератор случайных чисел для случайной расстановки ферзей
+# в задаче выше. Проверяйте различные случайные варианты и выведите 4 успешных расстановки.
+# *Выведите все успешные варианты расстановок
+
+from random import randint
+
+
+def check_position(pos: str) -> bool:
+    queens = [int(q) for q in pos.split()]
+    for i in range(len(queens)):
+        for j in range(i + 1, len(queens)):
+            # проверка на один ряд или одну колонку
+            if queens[j] // 10 == queens[i] // 10 or (queens[j] - queens[i]) % 10 == 0:
+                return False
+            # проверка на диагональ "вправо-вниз"
+            if ((queens[j] // 10 > queens[i] // 10 and queens[j] % 10 > queens[i] % 10) or
+                (queens[i] // 10 > queens[j] // 10 and queens[i] % 10 > queens[j] % 10)) and \
+                    (queens[j] - queens[i]) % 11 == 0:
+                return False
+            # проверка на диагональ "влево-вниз"
+            if ((queens[j] // 10 > queens[i] // 10 and queens[j] % 10 < queens[i] % 10) or
+                (queens[i] // 10 > queens[j] // 10 and queens[i] % 10 < queens[j] % 10)) and \
+                    (queens[j] - queens[i]) % 9 == 0:
+                return False
+    return True
```

Comment:
Условия верные, но можно значительно проще
```suggestion
    queens = [tuple(map(int, q)) for q in pos.split()]
    for i, (x0, y0) in enumerate(queens, 1):
        for x1, y1 in queens[i:]:
            if x0 == x1 or y0 == y1 or abs(x0 - x1) == abs(y0 - y1):
                return False

    return True
```